calicoctl/commands/node/diags.go

```diff
@@ -155,7 +159,39 @@ such as transfer.sh using curl or similar.  For example:
 
 }
 
-// writeDiags executes the dignostic commans and outputs the result in the file
+// getDockerLogs will attempt to grab logs for any "calico" named containers for hosted installs.
+func getDockerLogs(logDir string) {
+	os.Mkdir(logDir, os.ModeDir)
+
+	fmt.Println("Copying logs from Calico containers")
+	// Get a list of Calico containers running on this Node.
+	out, err := exec.Command("docker", "ps", "-a", "--filter", "\"name=calico\"", "--format", "{{.Names}}").Output()
+	if err != nil {
+		fmt.Printf("Could not run docker command: %s\n", err)
```

Comment:
It certainly isn't much effort to do, and could shed light on a possible permissions issue if the user isn't running as root.  As for the No Calico containers found it might not really be an error, as someone might not be running a hosted install.

Also, this should be short circuiting here, and it's not :(